.buildkite/premerge.steps.yaml

```diff
@@ -67,8 +67,6 @@ steps:
   - label: "generate-pipeline-steps"
     commands:
       - "ci/generate-and-upload-build-steps.sh"
-    env:
-        ENGINE_VERSION: "${ENGINE_VERSION}"
```

Comment:
I suppose this is actually passed in by default, right? This makes sense then. Why don't we do drop it everywhere? (see below comment)